jrt/src/com/yahoo/jrt/CryptoEngine.java

```diff
@@ -0,0 +1,17 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.jrt;
+
+
+import java.nio.channels.SocketChannel;
+
+
+/**
+ * Component responsible for wrapping low-level sockets into
+ * appropriate CryptoSocket instances. This is the top-level interface
+ * used by code wanting to perform network io with appropriate
+ * encryption.
+ **/
+public interface CryptoEngine {
+    public CryptoSocket createCryptoSocket(SocketChannel channel, boolean isServer);
```

Comment:
The signature should allow implementations to throw `IOException` if creating the socket fails.

Reply:
the socket is already created, and passed in as 'channel'

Reply:
ok, you meant the 'CryptoSocket' itself... yes, we can probably do that.

Reply:
Please note that the constructor may be called from another thread than the other methods